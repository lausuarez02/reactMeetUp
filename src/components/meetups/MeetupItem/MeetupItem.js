//hooks
import { useFetch} from "util-hooks/useFetch/useFetch";
//utils
import { useDispatch, useSelector } from 'react-redux'
import {useState ,useEffect} from "react";
import { useLocation } from "react-router-dom"
import classes from "./MeetupItem.module.css";
//components
import Card from 'components/ui/Card'
import { addToFavorites, removeFromFavorites } from "reduxStore/reducers/user/user";



export default function MeetupItem({image,title, address,description, id}) {
  const dispatch = useDispatch()
  const { data } = useFetch({
    url: "/data.json",
  });
  const pathName = useLocation().pathname;

  const test = useSelector((state) => state.userFavorites)

  const avoidDefaultAddFavorites = ({e, image,title, address,description, id}) => {
    e.preventDefault();
    dispatch(addToFavorites({image,title, address,description,id}))
  }

  const avoidDefaultRemoveFavorites = ({e, id}) => {
    e.preventDefault();
    dispatch(removeFromFavorites(id))

  }

  if (!data) return <p data-testid="component">Loading...</p>;

  return (
    <li className={classes.item} data-testid="component">
      
        {
           pathName === "/favorites" ? 

              <Card>
                    <div className={classes.image}>
                      <img src={image ? image : data.image} alt={title ? title : data.title} />
                    </div>
                    <div className={classes.content}>
                      <h3>{title ? title : data.title}</h3>
                      <address>{address ? address : data.address}</address>
                      <p>{description ? description : data.description}</p>
                    </div>
                    <div className={classes.actions}>
                     <button data-testid="RemoveFavorites" onClick={(e) => avoidDefaultRemoveFavorites({e:e,id: id})}>Remove from favorites</button>
                      
                    </div>
              </Card>

           :
           data.map((data) => {
            return(
            <Card key={data.id}>
                  <div className={classes.image}>
                    <img src={image ? image : data.image} alt={title ? title : data.title} />
                  </div>
                  <div className={classes.content}>
                    <h3>{title ? title : data.title}</h3>
                    <address>{address ? address : data.address}</address>
                    <p>{description ? description : data.description}</p>
                  </div>
                  <div className={classes.actions}>
                    <button data-testid="AddToFavorites" onClick={(e) => avoidDefaultAddFavorites({e:e, image: data.image, title: data.title, address: data.address,description: data.description, id:test.data.length})}>Add to favorites</button>
                    
                  </div>
            </Card>
            )
          })
        }
    </li>
  );
}
