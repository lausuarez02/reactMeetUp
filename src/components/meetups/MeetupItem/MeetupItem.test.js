/* eslint-disable testing-library/no-debugging-utils */
import React from 'react';
import { render, screen } from '@testing-library/react';
import MeetupItem from './MeetupItem';
import { Provider } from 'react-redux';
import configureStore from 'redux-mock-store';

describe('MeetupItem', () => {
  const mockStore = configureStore([]);

  it('renders the correct number of items', () => {
    const store = mockStore({
      userFavorites: {
        data: [
          {
            id: 1,
            title: 'Meetup 1',
            address: '123 Main St.',
            description: 'This is the first meetup',
            image: 'https://example.com/meetup1.jpg',
          },
          {
            id: 2,
            title: 'Meetup 2',
            address: '456 Main St.',
            description: 'This is the second meetup',
            image: 'https://example.com/meetup2.jpg',
          },
        ],
      },
    });

    render(
      <Provider store={store}>
        <MeetupItem />
      </Provider>
    );

    const items = screen.getAllByTestId('component');
    expect(items.length).toBe(2);
  });
});