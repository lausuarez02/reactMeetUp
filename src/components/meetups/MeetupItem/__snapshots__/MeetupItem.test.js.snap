// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Test cartBotton component test if data is on Meetup 1`] = `
<p
  data-testid="component"
>
  Loading...
</p>
`;
